ee/experimental/incident_utils.py

```diff
@@ -1,10 +1,152 @@
+        knee_threshold: float = 0.8,
+        min_incident_size: int = 5,
+        incident_similarity_threshold: float = 0.8,
+    ):
+    
+    if not incident_lower_timestamp:
+        incident_lower_timestamp = datetime.now() - timedelta(days=100)
+        
+    if not incident_upper_timestamp:
+        incident_upper_timestamp = datetime
+        
+    if not alert_lower_timestamp:
+        alert_lower_timestamp = datetime.now() - timedelta(days=100)
+        
+    if not alert_upper_timestamp:
+        alert_upper_timestamp = datetime.now()
+
+    calculate_pmi_matrix(ctx, tenant_id)
+
+    alerts = get_last_alerts(tenant_id, limit=use_n_historical_alerts, upper_timestamp=alert_upper_timestamp, lower_timestamp=alert_lower_timestamp)
+    incidents, _ = get_last_incidents(tenant_id, limit=use_n_hist_incidents, upper_timestamp=incident_upper_timestamp, lower_timestamp=incident_lower_timestamp)
+    nc_queue = NodeCandidateQueue()
+    
+    for candidate in [NodeCandidate(alert.fingerprint, alert.timestamp) for alert in alerts]:
+        nc_queue.push_candidate(candidate)
+    candidates = nc_queue.get_candidates()          
+    
+    graph = create_graph(tenant_id, [candidate.fingerprint for candidate in candidates], pmi_threshold, knee_threshold)
+    ids = []
+    
+    for component in nx.connected_components(graph):
+        if len(component) > min_incident_size:            
+            alerts_appended = False
+            for incident in incidents:
+                incident_fingerprints = set([alert.fingerprint for alert in incident.Incident.alerts])        
+                intersection = incident_fingerprints.intersection(component)
+        
+                if len(intersection) / len(component) >= incident_similarity_threshold:
+                    alerts_appended = True
+                    for alert in [alert for alert in alerts if alert.fingerprint in component]:
+                        if not is_alert_assigned_to_incident(alert.id, incident.Incident.id, tenant_id):
+                            assign_alert_to_incident(alert.id, incident.Incident.id, tenant_id)
+                    
+                    summary = generate_incident_summary(incident.Incident)
+                    update_incident_summary(incident.Incident.id, summary)
+                    
+            if not alerts_appended:
+                incident = create_incident_from_dict(tenant_id, {"name": "New Incident", "description": "Summorization is Disabled", "is_predicted": True})
+                ids.append(incident.id)
+                for alert in [alert for alert in alerts if alert.fingerprint in component]:
+                    if not is_alert_assigned_to_incident(alert.id, incident.id, tenant_id):
+                        assign_alert_to_incident(alert.id, incident.id, tenant_id)
+                    
+                summary = generate_incident_summary(incident)
+                update_incident_summary(incident.id, summary)
```

Comment:
what's hapenning here? why would an incident have the name "New Incident" with this weird description? :P

Reply:
Well noticed, incident name doesn't get updated, will fix that. As for the weird description - this is a placeholder which will be displayed if there'a something broken on the summary generation side

Reply:
Do you have any proposals on how an incident name should look like?